cleanup/artifactCleanup/artifactCleanup.groovy

```diff
@@ -39,16 +42,20 @@ class Global {
 // curl -i -uadmin:password -X POST "http://localhost:8081/artifactory/api/plugins/execute/cleanupCtl?params=command=stop"
 // curl -i -uadmin:password -X POST "http://localhost:8081/artifactory/api/plugins/execute/cleanupCtl?params=command=adjustPaceTimeMS;value=-1000"
 
+
 def pluginGroup = 'cleaners'
+def config = new ConfigSlurper().parse(new File(ctx.artifactoryHome.haAwareEtcDir, PROPERTIES_FILE_PATH).toURL())
 
 executions {
     cleanup(groups: [pluginGroup]) { params ->
         def months = params['months'] ? params['months'][0] as int : 6
         def repos = params['repos'] as String[]
-        def dryRun = params['dryRun'] ? params['dryRun'][0] as boolean : false
-        def disablePropertiesSupport = params['disablePropertiesSupport'] ? params['disablePropertiesSupport'][0] as boolean : false
+        def dryRun = params['dryRun'] ? params['dryRun'][0].toBoolean() : false      
+        def disablePropertiesSupport = params['disablePropertiesSupport'] ? params['disablePropertiesSupport'][0].toBoolean() : false
         Global.paceTimeMS = params['paceTimeMS'] ? params['paceTimeMS'][0] as int : 0
-        artifactCleanup(months, repos, log, Global.paceTimeMS, dryRun, disablePropertiesSupport)
+        def keepRelease = params['keepRelease'] ? params['keepRelease'][0].toBoolean() : false
+        def releaseRegex = config.policies[0][7] ? config.policies[0][7] as Pattern : ~/.*-\d+\.\d+\.\d+\.*/
```

Comment:
You are using the first policy regex as a default value for cleanup requests. Why aren't you getting the value from the parameters like the others?

Reply:
I think it is not possible to properly pass all regex chars in POST request (I think some of them are forbidden) so it would be easier to avoid this.

Reply:
You can percent-encode them by hand of course, but that's probably not the best approach. Still, just going with the first configured regex seems weird as well. Who's to say the first regex is the correct one? If you need one of the others (or an entirely different non-configured one) you're just out of luck ...

It might be possible to choose the "most correct" regex based on the provided repo names, or even their package types, but there are problems with that too.